hyperspy/misc/axis_tools.py

```diff
@@ -29,21 +29,19 @@ def check_axes_calibration(ax1, ax2):
 
     Parameters
     ----------
-    ax1, ax2 : DataAxis objects
+    ax1, ax2 : Axis objects
+        Axes objects that should be compared.
+    rtol : float
+        Tolerance passed to `np.allclose` for comparison. Default 1e-7.
 
     Returns
     -------
     bool
         If the two axes have identical calibrations.
 
     """
-    if not np.allclose(ax1.scale, ax2.scale, atol=0, rtol=1e-7):
-        return False
-
-    if not np.allclose(ax1.offset, ax2.offset, atol=0, rtol=1e-7):
-        return False
-
-    if ax1.units != ax2.units:
-        return False
-
-    return True
+    if ax1.size == ax2.size:
+        if np.allclose(ax1.axis, ax2.axis, atol=0, rtol=rtol) and\
+            ax1.units == ax2.units:
```

Comment:
In case the units are the same but different string (`s` versus `second` for example), is it worth using pint to compare the units here?

Reply:
Why not ;-)